packages/sacado/test/TestSuite/FadUnitTests.hpp

```diff
@@ -318,7 +377,7 @@ class FadOpsUnitTest : public CppUnit::TestFixture {
 template <class FadType, class ScalarType>
 FadOpsUnitTest<FadType,ScalarType>::
 FadOpsUnitTest() :
-  urand(), n(5), tol_a(1.0e-15), tol_r(1.0e-14) {}
+  urand(), n(5), tol_a(1.0e-15), tol_r(1.0e-12) {}
```

Comment:
Do the tests only work for double types? Should this be something like:
```
tol = std::numeric_limits<ScalarType>::epsilon() * 100.0;
```

Reply:
The tests are only instantiated on double.  Something like that probably would be better, although it is unclear what the factor (100 in this case) should be in general.

Reply:
Perhaps a `Teuchos::ScalarTraits<ScalarType>::magnitude_type`? That's if ScalarType is supported by Teuchos' scalar traits, of course. Though one could "easily" extend some of those traits to a SIMD type too.